patches/net/minecraft/client/multiplayer/ClientPacketListener.java.patch

```diff
@@ -100,14 +100,18 @@
      }
  
      @Override
-@@ -1574,6 +_,11 @@
+@@ -1574,6 +_,15 @@
      public void handleUpdateRecipes(ClientboundUpdateRecipesPacket p_105132_) {
          PacketUtils.ensureRunningOnSameThread(p_105132_, this, this.minecraft);
          this.recipes = new ClientRecipeContainer(p_105132_.itemSets(), p_105132_.stonecutterRecipes());
 +
 +        // Neo: abuse recipe sync to overwrite fuel values with datamap values after their sync (tag update doesn't fire on initial sync and the constructor is too early)
 +        if (this.connectionType.isNeoForge()) {
 +            this.fuelValues = net.neoforged.neoforge.common.DataMapHooks.populateFuelValues(this.registryAccess, this.enabledFeatures);
++        } else {
++            // Notify client mods that they're connected to a Vanilla server, which will never give them recipe data
++            var event = new net.neoforged.neoforge.client.event.RecipesReceivedEvent(java.util.Set.of(), net.minecraft.world.item.crafting.RecipeMap.create(java.util.List.of()));
++            net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(event);
 +        }
```

Comment:
Would it not make sense to move this to something akin to `ClientHooks`? And then process the events and datamap stuff there?